avi/resource_avi_cloud.go

```diff
@@ -236,10 +243,104 @@ func ResourceAviCloudRead(d *schema.ResourceData, meta interface{}) error {
 	return err
 }
 
+// Verify cloudState is valid vcenter state to configure mgmt IP
+func isValidCloudState(cloudState string) bool {
+	switch cloudState {
+	case
+		"VCENTER_DISCOVERY_COMPLETE_NO_MGMT_NW",
+		"VCENTER_DISCOVERY_COMPLETE",
+		"VCENTER_DISCOVERY_WAITING_DC",
+		"VCENTER_DISCOVERY_ONGOING":
+		return true
+	}
+	return false
+}
+
+// Wait untill vcenter inventory gets completed to configure mgmt IP
+func waitForVcenterState(cloudUuid string, client *clients.AviClient, maxRetry int) error {
```

Comment:
Are you waiting for a specific state or waiting for it to get ready for configuration?

Reply:
waiting to get ready for configuration